src/Microsoft.AspNetCore.Server.Kestrel/Filter/Internal/FilteredStreamAdapter.cs

```diff
@@ -30,72 +24,39 @@ public class FilteredStreamAdapter : IDisposable
             SocketInput = new SocketInput(memory, threadPool, bufferSizeControl);
             SocketOutput = new StreamSocketOutput(connectionId, filteredStream, memory, logger);
 
-            _connectionId = connectionId;
-            _log = logger;
             _filteredStream = filteredStream;
-            _memory = memory;
         }
 
-        public SocketInput SocketInput { get; private set; }
+        public SocketInput SocketInput { get; }
 
-        public ISocketOutput SocketOutput { get; private set; }
-
-        public Task ReadInputAsync()
-        {
-            _block = _memory.Lease();
-            // Use pooled block for copy
-            return FilterInputAsync(_block).ContinueWith((task, state) =>
-            {
-                ((FilteredStreamAdapter)state).OnStreamClose(task);
-            }, this);
-        }
-
-        public void Abort()
-        {
-            _aborted = true;
-        }
+        public ISocketOutput SocketOutput { get; }
 
         public void Dispose()
         {
             SocketInput.Dispose();
         }
 
-        private async Task FilterInputAsync(MemoryPoolBlock block)
+        public async Task ReadInputAsync()
         {
             int bytesRead;
-            while ((bytesRead = await _filteredStream.ReadAsync(block.Array, block.Data.Offset, block.Data.Count)) != 0)
-            {
-                SocketInput.IncomingData(block.Array, block.Data.Offset, bytesRead);
-            }
-        }
-
-        private void OnStreamClose(Task copyAsyncTask)
-        {
-            _memory.Return(_block);
-
-            if (copyAsyncTask.IsFaulted)
-            {
-                SocketInput.AbortAwaiting();
-                _log.LogError(0, copyAsyncTask.Exception, "FilteredStreamAdapter.CopyToAsync");
-            }
-            else if (copyAsyncTask.IsCanceled)
-            {
-                SocketInput.AbortAwaiting();
-                _log.LogError("FilteredStreamAdapter.CopyToAsync canceled.");
-            }
-            else if (_aborted)
-            {
-                SocketInput.AbortAwaiting();
-            }
 
-            try
-            {
-                SocketInput.IncomingFin();
-            }
-            catch (Exception ex)
+            do
             {
-                _log.LogError(0, ex, "FilteredStreamAdapter.OnStreamClose");
-            }
+                var block = SocketInput.IncomingStart();
```

Comment:
They did matter when we used to call `SocketInput.IncomingFin();` in `OnSocketClosed()`. We no longer do, so we can simplify.